lib/ui/painting/codec.cc

```diff
@@ -281,6 +349,18 @@ void InstantiateImageCodec(Dart_NativeArguments args) {
   const float decodedCacheRatioCap =
       tonic::DartConverter<float>::FromDart(Dart_GetNativeArgument(args, 3));
 
+  int maxWidth =
+      tonic::DartConverter<int>::FromDart(Dart_GetNativeArgument(args, 4));
+  if (maxWidth < 0) {
```

Comment:
Passing -1 and then swapping to the max int feels bad

Reply:
100% agree, will explore ways to pass in an optional value.